env0/data_template.go

```diff
@@ -122,14 +122,14 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
 	d.Set("project_ids", template.ProjectIds)
 	d.Set("terraform_version", template.TerraformVersion)
 	d.Set("ssh_keys", template.SshKeys)
-	if template.Retry.OnDeploy != nil {
+	if (template.Retry.OnDeploy != client.TemplateRetryOn{}) {
```

Comment:
Does this work? Doesn't it compare by ref and this would never be equal?
I guess you should test it locally if you already touched this place 🙏

Reply:
The provider tests cover this, since there is no drift when doing an update